peachjam/signals.py

```diff
@@ -189,14 +189,16 @@ def password_reset_customerio(sender, request, user, **kwargs):
     get_customerio().track_password_reset(user)
 
 
-@receiver(signals.post_save, sender=Judgment)
-def judgment_saved_generate_summary(sender, instance, created, **kwargs):
-    """Generate AI summary for a judgment when it's saved."""
-    # from peachjam.tasks import generate_judgment_summary
-
-    if (
-        not instance.summary
-        and not instance.summary_ai_generated
-        and (not instance.must_be_anonymized or instance.anonymized)
-    ):
-        generate_judgment_summary(instance.pk)
+@receiver(signals.post_save, sender=DocumentContent)
+def judgment_content_changed_generate_summary(sender, instance, **kwargs):
+    if not instance.document.doc_type == "judgment":
+        return
+    judgment = instance.document
+    should_generate = (
+        not judgment.summary  # No summary at all
+        or judgment.summary_ai_generated  # Summary exists but is AI-generated
+    ) and (
+        not judgment.must_be_anonymised  # Anonymization OK
```

Comment:
```suggestion
        not judgment.must_be_anonymised or judgment.anonymised  # Anonymization OK
```

Reply:
it either doesn't require anonymisation, or has been anonymised

Reply:
is this correct? not all judgments require anonymisation